File: 21. Pass pointer to function modify original.py
"""
Pass pointer to function – modify original variable (use list)
"""
def add_one(p):
    p[0] += 1

def main():
    x = [5]
    add_one(x)
    print(x[0])

if __name__ == '__main__':
    main()

# Time Complexity: O(1).
# Space Complexity: O(1).

File: 21. Pass pointer to function modify original.cpp
// Topic: Pass pointer to function – modify original variable
#include <iostream>
using namespace std;

void addOne(int* p)
{
    (*p)++;
}

int main()
{
    int x = 5;
    addOne(&x);
    cout << x << '\n';
    return 0;
}

// Time Complexity: O(1).
// Space Complexity: O(1).
